src/test/java/com/puppycrawl/tools/checkstyle/internal/ImmutabilityTest.java

```diff
@@ -0,0 +1,433 @@
+        "com.puppycrawl.tools.checkstyle.checks.naming.ConstantNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.InterfaceTypeParameterNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.LambdaParameterNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.LocalFinalVariableNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.LocalVariableNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.MemberNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.MethodNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.MethodTypeParameterNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.ParameterNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.PatternVariableNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.RecordComponentNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.RecordTypeParameterNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.StaticVariableNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.TypeNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.whitespace.TypecastParenPadCheck",
+        // Failing 'testClassesWithoutImmutableFieldsAreStateful'
+        "com.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck"
+    );
+
+    /**
+     * ArchCondition for immutable fields.
+     */
+    private static final ArchCondition<JavaField> BE_IMMUTABLE = new ImmutableFieldArchCondition();
+
+    /**
+     * Map fo module full name to module details.
+     */
+    private static final Map<String, ModuleDetails> MODULE_DETAILS_MAP =
+        XmlMetaReader.readAllModulesIncludingThirdPartyIfAny().stream()
+            .collect(Collectors.toMap(ModuleDetails::getFullQualifiedName,
+                                      moduleDetails -> moduleDetails));
+
+    /**
+     * Test to ensure that fields in util classes are immutable.
+     *
+     * @noinspection JUnitTestMethodWithNoAssertions
+     * @noinspectionreason JUnitTestMethodWithNoAssertions - asserts in callstack,
+     *     but not in this method
+     */
+    @Test
+    public void testUtilClassesImmutability() {
+        final JavaClasses utilClasses = new ClassFileImporter()
+            .withImportOption(ImportOption.Predefined.DO_NOT_INCLUDE_TESTS)
+            .importPackages("com.puppycrawl.tools.checkstyle.utils",
+                            "com.puppycrawl.tools.checkstyle.checks.javadoc.utils");
+
+        final DescribedPredicate<JavaField> zeroSizeArrays =
+            new DescribedPredicate<>("zero size arrays") {
+                @Override
+                public boolean apply(JavaField input) {
+                    return ZERO_SIZE_ARRAY_FIELDS.contains(input.getFullName());
+                }
+            };
+
+        final ArchCondition<JavaField> beSuppressed = new ArchCondition<>("be suppressed") {
```

Comment:
If all `beSuppressed` conditions are basically the same (other than the collection we are checking), can we just make some static method where we pass in the name of the collection to build these conditions? 

`final  ArchCondition<JavaField> beSuppressed = getSuppressionCondition(SUPPRESSED_FIELDS_IN_UTIL_CLASSES);`

Or, perhaps even more generically, we pass in a Predicate (java, not archunit).

`final  ArchCondition<JavaField> beMessageKey = getSuppressionConditionByPredicate(it -> it.getName().startsWith("MSG"));`

Reply:
Done, made a separate class, didn't make a generic (`<T>`) one as it wasn't looking neat.